tests/unit-tests/test_astra_embedding_retrieval.py

```diff
@@ -0,0 +1,112 @@
+
+    Coral reefs, often referred to as the "rainforests of the sea," are another critical habitat. They are bustling with life and serve as a vital ecosystem for many marine species. Corals themselves are fascinating organisms. They are made up of thousands of tiny creatures called polyps and have a symbiotic relationship with algae, which provides them with food through photosynthesis.
+
+    The diversity of marine life is vast, and each species plays a crucial role in the ocean's ecosystem. From the microscopic plankton that form the base of the oceanic food chain to the majestic blue whale, the largest animal to have ever lived on Earth, marine animals are an integral part of our world's biodiversity. Protecting these creatures and their habitats is essential for maintaining the health of our oceans and the planet as a whole.
+    """
+
+    # Define the desired chunk size and overlap size
+    chunk_size = 450
+    overlap_size = 50
+
+    # Function to generate chunks with the specified size and overlap
+    def generate_chunks(text, chunk_size, overlap_size):
+        chunks = []
+        start = 0
+        end = chunk_size
+        while start < len(text):
+            # If this is not the first chunk, move back 'overlap_size' characters to create the overlap
+            if start != 0:
+                start -= overlap_size
+            chunks.append(text[start:end])
+            start = end
+            end += chunk_size
+        return chunks
+
+    # Generate the chunks based on the narrative
+    chunks = generate_chunks(narrative, chunk_size, overlap_size)
+
+    # Output the first few chunks to ensure they meet the specifications
+    for i, chunk in enumerate(chunks[:3]):  # Displaying the first 3 chunks for brevity
+        print(f"Chunk {i+1}:\n{chunk}\n{'-'*50}\n")
+
+    title = "Marine Animals habitat"
+    # colbert stuff starts
+
+
+    colbert = ColbertTokenEmbeddings(
+        doc_maxlen=220,
+        nbits=1,
+        kmeans_niters=4,
+    )
+
+    passageEmbeddings = colbert.embed_documents(texts=chunks, title=title)
+
+    print(f"passage embeddings size {len(passageEmbeddings)}")
+
+    for pEmbd in passageEmbeddings:
+
+        print(f"passage embedding title {pEmbd.title()} size {len(pEmbd.get_all_token_embeddings())}")
+        print(f"passagen embedding id {pEmbd.id()}")
+        for tokenEmbd in pEmbd.get_all_token_embeddings():
+            print(f"    token embedding id {tokenEmbd.id} parent {tokenEmbd.parent_id} size {len(tokenEmbd.get_embeddings())}")
+
+
+    # Fetch the Base64 encoded string from the environment variable
+    encoded_zip = os.getenv('COLBERT_ASTRA_SCB')
```

Comment:
We cannot rely on an astra database like this in the CI

we should use cassio, and run the integration tests after the db creation in _run_e2e_tests.yaml file

additionally, to make it easier, we can spin up a docker container with dse. You can copy and paste this class
https://github.com/datastax/ragstack-ai/blob/main/ragstack-e2e-tests/e2e_tests/test_utils/cassandra_container.py